File: api/src/main/java/com/fundmate/api/service/BudgetService.java
package com.fundmate.api.service;

import com.fundmate.api.dto.request.BudgetRequest;
import com.fundmate.api.dto.response.BudgetResponse;
import com.fundmate.api.model.Budget;

import java.util.List;

public interface BudgetService {
    BudgetResponse createBudget(BudgetRequest budgetRequest);
    List<BudgetResponse> getBudgetsByAccountId(Long accountId);
    BudgetResponse getBudgetById(Long id);
    void deleteBudget(Long id);
}

File: frontend/src/services/budget.service.js
import api from "./api";

export const budgetService = {
  async getBudgetsByAccount(accountId) {
    const response = await api.get(`/budgets/account/${accountId}`);
    return response.data;
  },

  async createBudget(budgetData) {
    const response = await api.post("/budgets", budgetData);
    return response.data;
  },

  async getBudgetById(id) {
    const response = await api.get(`/budgets/${id}`);
    return response.data;
  },

  async deleteBudget(id) {
    const response = await api.delete(`/budgets/${id}`);
    return response.data;
  },
};
